### PR TITLE
SYL-152 - Bypass new payment creation when we abord the old payment

### PR DESCRIPTION
**Pull request by TheGrimmChester**

### CODE REVIEW

File: src/Action/CaptureAction.php [hunk 1]
```diff
@@ -200,15 +196,24 @@ private function addNotificationUrl(TokenInterface $token, ArrayObject $details)
...
+     * @throws UnknownApiErrorException
+     * @throws BadRequestException
+     */
+    private function createPayment(ArrayObject $details, Capture $request): Payment
     {
         try {
+            if ($details->offsetExists('payment_id')
+                && $details->offsetExists('status')) {
+                $this->abortPaymentProcessor->process($request->getFirstModel());
+                unset($details['status'], $details['payment_id'], $details['is_live']);
```
Comment (oallain) [inline src/Action/CaptureAction.php:214]: Why did you unset 'is_live' ?

File: src/Handler/PaymentNotificationHandler.php [hunk 2]
```diff
@@ -72,9 +72,7 @@ public function treat(Generic $request, IVerifiableAPIResource $paymentResource,
 
         $this->entityManager->refresh($request->getFirstModel());
 
-        if (PayPlugApiClientInterface::STATUS_CREATED === $request->getFirstModel()->getDetails()) {
-            $lock->release();
-
+        if ($details['status'] === PayPlugApiClientInterface::STATUS_ABORTED) {
```
Comment (oallain) [inline src/Handler/PaymentNotificationHandler.php:75]: Yon don't need to release the lock here ?

Comment (TheGrimmChester) [reply src/Handler/PaymentNotificationHandler.php:75]: added it back, but it should be released on class destruct anyway

File: src/Action/CaptureAction.php (hunk 1 above)
Comment (TheGrimmChester) [reply src/Action/CaptureAction.php:214]: payplug is the data master for these fields, so we just update them using their response after the payment creation

Comment (oallain) [reply src/Action/CaptureAction.php:214]: Ok, i understand.

**Review by oallain — APPROVED** [review]

**Pull request merged**